### PR TITLE
Add RAPIDS::API service

### PR DESCRIPTION
**Pull request by FerPerales**
[Asana ticket](https://app.asana.com/0/1203289004376659/1206498816477316/f)

This is the first part of the feature. I decided to split this into several PRs.

* Add oauth2 gem 
* Create RAPIDS models for OccupationStandard and Organization
* Add RegistrationAgency.registration_agency_for_national_program 
* Add RAPIDS::API service to connect to RAPIDS API
* Create factories for API responses

### CODE REVIEW

File: app/models/rapids/occupation_standard.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+          parsed_sponsor_name = sponsor_number.match(/-?(?<state>[A-Z]{2})/)
+
+          state_abbreviation = parsed_sponsor_name && parsed_sponsor_name[:state]
+
+          state = State.find_by(
+            abbreviation: state_abbreviation
+          )
+
+          if state
+            registration_agency = state.registration_agencies.last
```
Comment (littleforest) [inline app/models/rapids/occupation_standard.rb:41]: Some states have both OA and SAA -- do we know which one we would want to grab in those cases?

**Review by littleforest — COMMENTED** [review]
Looks good to me, just one question.

File: app/models/rapids/occupation_standard.rb (hunk 1 above)
Comment (FerPerales) [reply app/models/rapids/occupation_standard.rb:41]: @littleforest didn't know that case was possible 😮  I'll ask Patrick

Comment (FerPerales) [reply app/models/rapids/occupation_standard.rb]: It must use OA. Code updated to reflect this

**Pull request merged**